### PR TITLE
Update ovh flavors

### PR DESCRIPTION
**Pull request by kshtsk**
Add s1 flavor to make teuthology working in new regions in OVH
Add workaround for floating ip exception for some regions in OVH

### CODE REVIEW

**Comment by kshtsk** [comment]
Related issues:
https://github.com/SUSE/teuthology/issues/77
https://github.com/SUSE/teuthology/issues/108

File: teuthology/openstack/__init__.py [hunk 1]
```diff
@@ -963,7 +970,14 @@ def get_unassociated_floating_ip():
...
         """
-        ips = json.loads(OpenStack().run("ip floating list -f json"))
+        try:
+            ips = json.loads(OpenStack().run("ip floating list -f json"))
+        except subprocess.CalledProcessError as e:
+            log.warning(e)
+            if e.returncode == 1:
+                return None
+            else:
+                raise e
```
Comment (smithfarm) [inline teuthology/openstack/__init__.py]: When it was just one line [1] being duplicated it wasn't a big deal. But now that you add this try..except block the amount of code being duplicated increases considerably. Can you split this out into a helper method to eliminate the code duplication?

[1] ips = json.loads(OpenStack().run("ip floating list -f json"))

Comment (kshtsk) [reply teuthology/openstack/__init__.py]: agree

File: teuthology/openstack/__init__.py [hunk 2]
```diff
@@ -454,7 +465,7 @@ def __flavor_wrapper(self, min, good, hint, arch):
         select = None
         if self.get_provider() == 'ovh':
             log.debug("Looking for a match among the El Cheapo flavors...")
-            select = '^vps-ssd-'
+            select = '^(vps-ssd|s1)-'
```
Comment (smithfarm) [inline teuthology/openstack/__init__.py]: @kshtsk Did you want to reverse the order here?

Comment (kshtsk) [reply teuthology/openstack/__init__.py]: @smithfarm yes... probably this can be implemented with ECP feature.

Comment (smithfarm) [reply teuthology/openstack/__init__.py]: @kshtsk Good idea. Then this PR could be closed.

**Comment by kshtsk** [comment]
#124 is covering this request

**Pull request closed without merging**